src/codewhisperer/commands/basicCommands.ts

```diff
@@ -120,37 +121,43 @@ export const showSsoSignIn = Commands.declare('aws.codeWhisperer.sso', () => asy
 })
 
 // Shortcut command to directly connect to Identity Center or prompt start URL entry
-// It can optionally set a customization too.
+// It can optionally set a customization too based on given values to match on
 export const connectWithCustomization = Commands.declare(
     'aws.codeWhisperer.connect',
-    () =>
-        async (
-            startUrl?: string,
-            region?: string,
-            customizationArn?: string,
-            customizationName?: string,
-            customizationDescription?: string
-        ) => {
-            // This command supports two sets of arguments:
-            //  * startUrl and region. If both arguments are provided they will be used, otherwise
-            //    the command prompts for them interactively.
-            //  * customization{Arn, Name, Description}. If at least customizationArn is provided,
-            //    the command selects this customization.
-            if (startUrl && region) {
-                await connectToEnterpriseSso(startUrl, region)
-            } else {
-                await getStartUrl()
-            }
-            if (customizationArn) {
-                const match = getPersistedCustomizations().find(c => c.arn == customizationArn)
-                const customization = {
-                    arn: customizationArn,
-                    name: customizationName ?? match?.name ?? 'unknown',
-                    description: customizationDescription ?? match?.description ?? 'unknown',
-                }
-                await selectCustomization(customization)
-            }
+    () => async (startUrl?: string, region?: string, customizationArn?: string, customizationNamePrefix?: string) => {
+        // This command supports two sets of arguments:
+        //  * startUrl and region. If both arguments are provided they will be used, otherwise
+        //    the command prompts for them interactively.
+        //  * customization{Arn, Name}. Selects a customization based on whether a match for given values are found.
+        if (startUrl && region) {
+            await connectToEnterpriseSso(startUrl, region)
+        } else {
+            await getStartUrl()
+        }
+
+        // No customization match information given, exit early.
+        if (!customizationArn && !customizationNamePrefix) {
+            return
         }
+
+        let match: Customization | undefined
```

Comment:
this can be const, because of the condition on line 139.

```suggestion
        const match = customizationArn ? ... : ...
```